torchdynamo/eval_frame.py

```diff
@@ -240,12 +240,23 @@ def toy_example(a, b):
 
 
 def export(f, *args, **kwargs):
-    from inspect import signature
-
     import torch.utils._pytree as pytree
 
     graph = None
     out_guards = None
+    compiler_captured_inputs = None
+
+    def produce_matching(source_args, candidate_args):
+        matched_elements_positions = []
+        for x in candidate_args:
+            matched_elements = [torch.allclose(x, y) for y in source_args]
```

Comment:
This method of matching seems wrong.  `x is y` seems like the correct thing.  

We could have a function that returns two different tensors with the same elements in the example input.  (But not all inputs.)

Perhaps we need to only run the model once so the tensors will pass an `x is y` test?

Reply:
Agreed - but wasn't sure how we could run only once. We need to run it *once* to capture the graph - so that would mean that the second run here, the one where we run the exported graph, is optional. We could, ostensibly, instead of running the second graph, do analysis on the placeholder/input nodes and output of the function... I left a todo here, would you be open to kicking this can down the road?

Reply:
Resolved